### PR TITLE
ROC-6719 mapping mediation status

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-6719
https://tools.hmcts.net/jira/browse/ROC-6660

### Change description ###
Mediation Outcome with the Callback

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+        return AboutToStartOrSubmitCallbackResponse
+            .builder()
+            .data(dataMap)
+            .build();
+    }
+
+    private String stateByOnlineDQnPilotCheck(Claim claim) {
+
+        Predicate<Response> isPartAdmitOrFullDefence = response ->
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java]: DirectionQuestionnnaireUtils seems unrelated. So moved to ResponseHelper.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/MediationOutcomeMapper.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import static uk.gov.hmcts.cmc.domain.models.MediationOutcome.SUCCEEDED;
+
+public class MediationOutcomeMapper {
+
+    public static MediationOutcome from(CCDCase ccdCase){
+
+        CCDRespondent defendant = ccdCase.getRespondents()
+            .stream().findFirst().map(CCDCollectionElement::getValue).orElseThrow(IllegalStateException::new);
+
+        if(Optional.ofNullable(defendant.getMediationFailedReason()).isPresent()){
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/MediationOutcomeMapper.java]: Done

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.function.Predicate;
+
+import static uk.gov.hmcts.cmc.ccd.domain.CaseEvent.MEDIATION_FAILED;
+import static uk.gov.hmcts.cmc.claimstore.services.ccd.Role.CASEWORKER;
+import static uk.gov.hmcts.cmc.domain.models.claimantresponse.ClaimantResponseType.REJECTION;
+import static uk.gov.hmcts.cmc.domain.models.response.ResponseType.FULL_DEFENCE;
+import static uk.gov.hmcts.cmc.domain.models.response.ResponseType.PART_ADMISSION;
+
+@Service
+public class MediationFailedCallbackHandler extends CallbackHandler {
```
[the remark this answers is not in the document]
Comment (karthikkannankainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java]: MediationFailed is an event. So handling callback based on the event. The name is not generic as its specific to event. There is separate MediationSuccess events. Business wants it be specific.

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4396597&bid=14903998)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4396597&bid=14903998)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4396597&bid=14903998)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  7
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java  1
- ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/HMCTS/cmc-claim-store/pullRequest?prid=4396597&bid=14903998)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java  1
- ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4396597&bid=14903998)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandler.java  7
- ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationFailedCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4396597&bid=14903998)

**Pull request merged**